src/helics/common/JsonProcessingFunctions.cpp

```diff
@@ -16,15 +16,34 @@ SPDX-License-Identifier: BSD-3-Clause
 
 namespace helics::fileops {
 
-bool looksLikeCommandLine(std::string_view testString)
+bool looksLikeJson(std::string_view jsonString)
 {
-    if (testString.empty()) {
+    if (jsonString.empty()) {
         return false;
     }
-    if (testString.front() == '-') {
+    if ((jsonString.front() == '{') || (jsonString.front() == '//' && jsonString.back() == '}')) {
         return true;
     }
-    return testString.find(" -") != std::string::npos;
+    if (jsonString.front() == '#' || jsonString.find("\n#") != std::string_view::npos) {
+        return false;
+    }
+    auto firstColonLoc = jsonString.find_first_of(':');
+    if (firstColonLoc == std::string_view::npos) {
+        return false;
+    }
+    auto lastColonLoc = jsonString.find_last_of(':');
+    auto openBracket = jsonString.find_first_of('{');
+    if (openBracket == std::string_view::npos) {
+        return false;
+    }
+    auto closeBracket = jsonString.find_last_of('}');
+    if (closeBracket == std::string_view::npos) {
+        return false;
+    }
```

Comment:
I don't think either of these conditions is true. Valid JSON that would fail these checks include:

```json
[23, "abc", "def"]
```

```json
[]
```

```json
"abc"
```

```json
23
```

Reply:
while true, those examples would not be valid config files